particles/Tutorial/Javascript/Demo/source/Game.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+const resetMove = {type: 'reset', move: -1};
+const clearMove = {type: 'clear', move: -1};
+
+/* global defineParticle */
+defineParticle(({SimpleParticle, html, log}) => {
+
+  return class extends SimpleParticle {
+
+    get template() {
+      return html`
+      <span>It is your turn <span>{{name}}</span>, playing as <span>{{avatar}}</span>.
+      <div slotid="gameSlot"></div>
+      <div hidden={{hideCongrats}}>Congratulations <span>{{winnerName}}</span>!</div>
+    `;
+    }
+
+    shouldRender({gameState, playerOne, playerTwo}) {
+      return gameState && playerOne && playerTwo;
+    }
+
+    render({gameState, playerOneMove, playerTwoMove, playerOne, playerTwo, move, event}) {
+      const toReturn = {hideCongrats: true};
+      if (!move) {
+        this.set('move', resetMove);
+        return toReturn;
+      }
+
+      if (playerOne.id != 0) {
+        this.set('playerOne', {name: playerOne.name, avatar: playerOne.avatar, id: 0});
+        return toReturn;
+      }
+
+      if (!playerTwo.id) {
+        this.set('playerTwo', {name: playerTwo.name, avatar: playerTwo.avatar, id: 1});
+        return toReturn;
+      }
+
+      // If the move has been applied, and the game isn't over
+      if (gameState.lastMove == move.move && !gameState.gameOver) {
+
+        if (gameState.currentPlayer == 0) {
```

Comment:
It's best to keep `render` and `update` as short as possible. I tend to do my conditionals in these methods, but then factor actual work into functions. That way these state-managing functions tend to express the author's decision making logic (and the nitty gritty is elsewhere).